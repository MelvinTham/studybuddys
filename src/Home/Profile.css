.profile {
    position: relative;
    padding: 60px 70px;
    width: 60%;
    display: grid;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin:20px;
}

/* 
.event {
    position: relative;
    display: inline-block;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin: 10px;
} */