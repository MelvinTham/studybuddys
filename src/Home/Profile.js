import './Profile.css';
import React, { PureComponent } from 'react';
import {
    TextField,
    Typography,
    Card,
    Button,
    Paper
} from '@material-ui/core';
import axios from 'axios';
import * as firebase from 'firebase';
import "./Dashboard.css";

class Profile extends PureComponent{
    constructor(props) {
        super(props);
            this.state = {
                userClass: "",
                name: "",
                password: "",
                email: "",
                facebook: "facebook-link",
                linkedIn: "linkedIn-link",
                instagram: "instagram-link",
                eventName: ["Study With Jenny", "Lunch At Cafe Amore", "Study for Paradigms Final"],
                description: ["One on One Date", "Food is Important", "Kill me now"],
                dateAndTime: ["Dec 06, 2018 3:30PM", "Dec 06, 2018 4:30PM", "Dec 16 2018, 12:00PM"]
            }
        this.getUserName = this.getUserName.bind(this);
        this.changeEmail = this.changeEmail.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.changePassword = this.changePassword.bind(this);
    }

    componentDidMount(){
        this.getUserName();
    }

    handleChange = name => event => {
        this.setState({
            [name]: event.target.value,
        });
    };

    changeEmail(e){
        var user = firebase.auth().currentUser;
        var email = this.state.email;
        user.updateEmail(email).then(function() {
            alert("Successful update");
        }).catch(function(error) {
            alert(error);
        });
    }

   changePassword(e){
        var user = firebase.auth().currentUser;
        var password = this.state.pass
        user.updatePassword(password).then(function() {
            alert("Successful update");
        }).catch(function(error) {
            alert(error);
        });
    }

    getUserName(e){
        var page = this;
        firebase.auth().onAuthStateChanged(function(user) {
            if (user) {
                var info = {
                id: user.uid
                }
                axios.post('https://studybuddys-223920.appspot.com/api/getUsername', info)
                .then(response => {
                page.setState({name : response.data.name})
                })
            }
        });
    }

    render(){

        return(
            <div>

                <div
                    className = "flexCenter"
                    raised = {true}
                    data-aos="fade-in"
                    data-aos-easing="linear"
                    data-aos-duration="800"
                >

                    <Typography variant = "h3"><br></br><u>Profile</u></Typography>
                    <br/>
                    <Card
                        raised = {true}
                        className = "profile"
                        data-aos="fade-in"
                        data-aos-easing="linear"
                        data-aos-duration="800"
                    >
                        <Typography variant = "h4">Your Current Information:</Typography>
                        <br/>
                        <Typography variant = "h6">Name: {this.state.name}</Typography>
                        <Typography variant = "h6">Facebook: {this.state.facebook}</Typography>
                        <Typography variant = "h6">LinkedIn: {this.state.linkedIn}</Typography>
                        <Typography variant = "h6">Instagram: {this.state.instagram}</Typography>
                    </Card>

                    <Card
                        raised = {true}
                        className = "profile"
                        data-aos="fade-in"
                        data-aos-easing="linear"
                        data-aos-duration="800"
                    >
                        <Typography variant = "h4">Update Information:</Typography>
                        <br/>
                        <TextField
                            style = {{marginTop:'18px'}}
                            type='email'
                            onChange = {this.handleChange("email")}
                            placeholder = "Email Address"
                        />
                        <Button onClick = {this.changeEmail}>Save Change</Button>
                        <TextField
                            style = {{marginTop:'18px'}}
                            onChange = {this.handleChange("pass")}
                            placeholder = "Password"
                        />
                        <Button onClick = {this.changePassword}>Save Change</Button>
                    </Card>

                    <Card
                        raised = {true}
                        className = "profile"
                        data-aos="fade-in"
                        data-aos-easing="linear"
                        data-aos-duration="800"
                    >
                        <Typography variant = "h4">Social Media Accounts</Typography>
                        <br/>
                        <TextField
                            type = "url"
                            placeholder = "Facebook"
                            style = {{marginTop: "18px"}}
                        />
                        <Button>Save Change</Button>
                        <TextField
                            type = "url"
                            placeholder = "LinkedIn"
                            style = {{marginTop: "18px"}}
                        />
                        <Button>Save Change</Button>
                        <TextField
                            type = "url"
                            placeholder = "Instagram"
                            style = {{marginTop: "18px"}}
                        />
                        <Button>Save Change</Button>
                    </Card>

                    <Card
                        raised = {true}
                        className = "profile"
                        data-aos="fade-in"
                        data-aos-easing="linear"
                        data-aos-duration="800"
                    >
                        <Typography variant = "h4">Upcoming Events</Typography>
                        <br/>
                        {this.state.eventName.map((title, key) => {
                            return(
                                <Paper className="event">
                                    {title}
                                    <br/>
                                    {this.state.description[key]}
                                    <br/>
                                    {this.state.dateAndTime[key]}
                                </Paper>
                            )
                        })}
                    </Card>
                </div>
            </div>
        )
    }
}
export default Profile;
