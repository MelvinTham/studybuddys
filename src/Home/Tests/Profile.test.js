import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import Profile from './../Profile';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('Profile component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <Profile />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <Profile />
        </Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
