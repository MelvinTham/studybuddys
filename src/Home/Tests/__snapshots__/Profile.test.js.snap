// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component render snapshot 1`] = `
<div>
  <div
    class="nav"
  >
    <div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="float:left"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-31 MuiSvgIcon-fontSizeLarge-39"
            focusable="false"
            role="presentation"
            style="color:white"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="flexRow"
      style="margin-left:auto"
    >
      <span
        style="font-size:22px;margin-right:10px"
      >
        <a
          href="/dashboard/profile"
          style="color:white"
        />
      </span>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        style="color:white;font-size:17px"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Logout
        </span>
      </button>
    </div>
  </div>
  <div
    class="flexCenter"
    data-aos="fade-in"
    data-aos-duration="800"
    data-aos-easing="linear"
  >
    <h3
      class="MuiTypography-root-54 MuiTypography-h3-68"
    >
      <br />
      <u>
        Profile
      </u>
    </h3>
    <br />
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Your Current Information:
      </h4>
      <br />
      <h6
        class="MuiTypography-root-54 MuiTypography-h6-71"
      >
        <strong>
          Name:
        </strong>
         
      </h6>
      <h6
        class="MuiTypography-root-54 MuiTypography-h6-71"
      >
        <strong>
          Email:
        </strong>
         
      </h6>
      <h6
        class="MuiTypography-root-54 MuiTypography-h6-71"
      >
        <strong>
          Facebook:
        </strong>
         Facebook
      </h6>
      <h6
        class="MuiTypography-root-54 MuiTypography-h6-71"
      >
        <strong>
          LinkedIn:
        </strong>
         LinkedIn
      </h6>
      <h6
        class="MuiTypography-root-54 MuiTypography-h6-71"
      >
        <strong>
          Instagram:
        </strong>
         Instagram
      </h6>
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Update Account Information:
      </h4>
      <br />
      <div
        class="MuiFormControl-root-117"
        style="margin-top:18px"
      >
        <div
          class="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
            placeholder="Email Address"
            type="email"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Save Change
        </span>
      </button>
      <div
        class="MuiFormControl-root-117"
        style="margin-top:18px"
      >
        <div
          class="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Save Change
        </span>
      </button>
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Social Media Accounts
      </h4>
      <h6
        class="MuiTypography-root-54 MuiTypography-subtitle1-72"
      >
        (Leave As Is If You're Not Updating)
      </h6>
      <br />
      <div
        class="MuiFormControl-root-117"
        style="margin-top:18px"
      >
        <div
          class="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
            type="url"
            value="facebook.com"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root-117"
        style="margin-top:18px"
      >
        <div
          class="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
            type="url"
            value="linkedin.com"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root-117"
        style="margin-top:18px"
      >
        <div
          class="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
            type="url"
            value="instagram.com"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-3"
        >
          Save Change(s)
        </span>
      </button>
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Please Respond Events
      </h4>
      <br />
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Awaiting Response Events
      </h4>
      <br />
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Approved Events
      </h4>
      <br />
    </div>
    <div
      class="MuiPaper-root-90 MuiPaper-elevation8-100 MuiPaper-rounded-91 MuiCard-root-89 profile"
      data-aos="fade-in"
      data-aos-duration="800"
      data-aos-easing="linear"
    >
      <h4
        class="MuiTypography-root-54 MuiTypography-h4-69"
      >
        Denied Events
      </h4>
      <br />
    </div>
  </div>
</div>
`;
